twister: twister-outs limit 99 log files causes problems

**Describe the bug**
Twister implementation in rotating logs seems to be causing problems. When reached 99 twister-outs, twister starts malfunction. Editing code doesn't change anything and random errors appear. I attached log from some random error, after deleting twister-outs, this tests was working properly.

**To Reproduce**
Steps to reproduce the behavior:
1. create 99 twister-outs
2. run any test
3. change something in test
4. Output does not change, and sometimes, there are errors

**Expected behavior**
Older logs should rotate, and twister should be usable

**Impact**
annoyance

**Logs and console output**
th no effect [-Werror=unused-value]
   69 |                         printk
      |                         ^~~~~~
/home/zephyr/zephyrproject/zephyr/samples/hello_world_cpu_ticks/src/main.c:69:31: error: expected '; before '}' token
   69 |                         printk
      |                               ^
      |                               ;
   70 |                 }
      |                 ~  

**Environment (please complete the following information):**
 - Linux
 - Zephyr SDK 16.8
 - any

--- a/scripts/pylib/twister/twisterlib/twister_main.py
+++ b/scripts/pylib/twister/twisterlib/twister_main.py
@@ -94,6 +94,9 @@ def main(options, default_options):
                     print("Renaming output directory to {}".format(new_out))
                     shutil.move(options.outdir, new_out)
                     break
+            else:
+                sys.exit(f"Too many '{options.outdir}.*' directories. Run either with --no-clean, "
+                         "or --clobber-output, or delete these directories manually.")
 
     previous_results_file = None
     os.makedirs(options.outdir, exist_ok=True)

